zingolib/src/wallet/notes/interface.rs

```diff
@@ -0,0 +1,70 @@
+use incrementalmerkletree::{Hashable, Position};
+use zcash_primitives::{memo::Memo, merkle_tree::HashSer, transaction::TxId};
+
+use super::super::{
+    data::TransactionRecord,
+    keys::unified::WalletCapability,
+    traits::{FromBytes, FromCommitment, Nullifier, ReadableWriteable, ToBytes},
+    Pool,
+};
+
+pub trait NoteInterface: Sized {
+    fn spent(&self) -> &Option<(TxId, u32)>;
+    fn is_spent(&self) -> bool {
+        Self::spent(self).is_some()
+    }
+    fn spent_mut(&mut self) -> &mut Option<(TxId, u32)>;
+    fn pending_spent(&self) -> &Option<(TxId, u32)>;
+    fn pending_spent_mut(&mut self) -> &mut Option<(TxId, u32)>;
+}
+
+///   All zingolib::wallet::traits::Notes are NoteInterface
+///   NoteInterface provides...
```

Comment:
this doc comment is for NoteInterface not ShieldedNoteInterface.... also its not very helpful haha

i think everything in wallet::traits is for shielded right? or maybe the from bytes/ to bytes might be applicable to transprent im not sure

Reply:
i see this comment was made before you created NoteInterface... interesting..

maybe change to ShieldedNoteInterface?

Reply:
fixed in https://github.com/zingolabs/zingolib/pull/809